theme_compassion_2025/static/src/js/_progress_bar.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * A customizable progress bar widget with animated flowing stripes.
+ *
+ * Usage:
+ *
+ *   const ProgressBarWidget = publicWidget.registry.ProgressBarWidget;
+ *   const widget = new ProgressBarWidget(this, {
+ *       density: 'high',        // "low" | "medium" | "high" (controls stripe density)
+ *       flowSpeed: '3s',        // CSS duration for stripe animation (default 2s)
+ *       steps: [                // Array of messages shown during progress
+ *           "Step 1...",
+ *           "Step 2...",
+ *           "Step 3..."
+ *       ]
+ *   });
+ *   await widget.appendTo($("#some-container"));
+ *   widget.startProgress();     // starts the animation and step updates
+ */
+
+odoo.define("theme_compassion_2025.ProgressBarWidget", function (require) {
+    "use strict";
+
+    var publicWidget = require("web.public.widget");
+
+    var ProgressBarWidget = publicWidget.Widget.extend({
+        template: "theme_compassion_2025.ProgressBarComponent",
+        xmlDependencies: ["/theme_compassion_2025/static/src/xml/ProgressBar.xml"],
+
+        defaults: {
+            steps: [
+                "msg nr.1…",
+                "msg nr.2…",
+                "another msg…",
+                "the more msg there are…",
+                "the more slow the progress…",
+                "bar.",
+            ],
```

Comment:
Defaults should be sensible values suitable for most common scenarios IMO, not a usage example (which you can have in your comment at the top of the file)

```suggestion
            steps: [
                "Loading…",
            ],
```